pudb/debugger.py

```diff
@@ -486,6 +486,25 @@ def _runscript(self, filename):
         # Implicitly runs in the namespace of __main__.
         self.run(statement)
 
+    def _runmodule(self, module_name):
+        import runpy
+        mod_name, mod_spec, code = runpy._get_module_details(module_name)
+        self.mainpyfile = self.canonic(code.co_filename)
+        import __main__
+        __main__.__dict__.clear()
+        __main__.__dict__.update({
+            "__name__": "__main__",
+            "__file__": self.mainpyfile,
+            "__package__": mod_spec.parent,
+            "__loader__": mod_spec.loader,
+            "__spec__": mod_spec,
+            "__builtins__": __builtins__,
+        })
+
+        self._wait_for_mainpyfile = True
+
+        self.run(code)
```

Comment:
This bit was basically stolen from pdb's code: https://github.com/python/cpython/blob/3.8/Lib/pdb.py#L1530

Reply:
Could you add the provenance as a comment?

Reply:
Will do!
I also just realized this breaks in PY2, will fix that as well.